src/menu/main_menu.c

```diff
@@ -0,0 +1,307 @@
+                                           .p0_cb = NULL,
+                                           .ui_cb = main_menu_handler,
+                                           .usb_cb = NULL,
+                                           .nfc_cb = NULL,
+                                           .evt_cfg_ptr = &main_menu_evt_config,
+                                           .flow_data_ptr = NULL};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+// /**
+//  * @brief Get the selected wallet index object
+//  *
+//  * @return uint8_t Wallet index selected from the main menu
+//  */
+// static uint8_t get_selected_wallet_index(void);
+
+/**
+ * @brief Get the boolean flag which depicts whether rendering the main menu is
+ * required or not
+ *
+ * @return true If the update is required
+ * @return false If the update is NOT required
+ */
+static bool main_menu_get_update_req(void);
+
+/**
+ * @brief This function determines which menu option was selected from the main
+ * menu. This selection cannot be static logic as the number of wallets/ number
+ * of options on the main menu are dynamic.
+ *
+ * @param menu_selectn_idx The selection index from the main menu
+ * @return main_menu_options_e The option of type main_menu_options_e detected
+ * by the lookup function
+ */
+static main_menu_options_e main_menu_lookup(uint16_t menu_selectn_idx);
+
+/**
+ * @brief This function resets internal values of the main menu module
+ *
+ */
+static void main_menu_reset_context(void);
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+// static uint8_t get_selected_wallet_index(void) {
+//   return main_menu_ctx.wallet_selected;
+// }
+
+static bool main_menu_get_update_req(void) {
+  return main_menu_ctx.update_req;
+}
+
+static main_menu_options_e main_menu_lookup(uint16_t menu_selectn_idx) {
+  main_menu_options_e menu_selected = MAIN_MENU_NONE;
+
+  /* Check if the selected index is greater than the wallet count */
+  if ((0 == main_menu_ctx.wallet_count) ||
+      (main_menu_ctx.wallet_count < menu_selectn_idx)) {
+    /* Normalize the menu selection index by subtracting the wallet count */
+    menu_selectn_idx = menu_selectn_idx - main_menu_ctx.wallet_count;
+
+    /* Order of the main menu: WALLET-x -> MAIN_MENU_CREATE_WALLET ->
+     * MAIN_MENU_SETTINGS */
+    if (1 == menu_selectn_idx) {
+      menu_selected = MAIN_MENU_CREATE_WALLET;
+    } else if (2 == menu_selectn_idx) {
+      menu_selected = MAIN_MENU_SETTINGS;
+    }
```

Comment:
in case of full wallets, make sure clicking on `Settings` option does not takes us to `Create wallet`

Reply:
fixed this logic